{
 "metadata": {
  "name": "",
  "signature": "sha256:50814cc1227083a9a7da6649bcf007380d00f5296bd40b5e4cd8c3f2dac882cc"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import sys\n",
      "sys.path.append('../../src/')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 50
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "from helpers import *"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 51
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sample_files = !ls $local_data/nov_24_run/fastq"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 52
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "run_template_yaml = '../../pipeline/run_template.yaml'\n",
      "sample_template_yaml = '../../pipeline/sample_template.yaml'"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 53
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "!head $run_template_yaml"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "fc_date: experiment_date\r\n",
        "fc_name: experiment_name\r\n",
        "upload:\r\n",
        "  dir: output_dir\r\n",
        "details:\r\n"
       ]
      }
     ],
     "prompt_number": 54
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "!head $sample_template_yaml"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "  - files: [file1, file2] \r\n",
        "    description: sample_name\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n"
       ]
      }
     ],
     "prompt_number": 55
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "out_template = '../../pipeline/nov_24_run.yaml'\n",
      "run_info = {}\n",
      "run_info['experiment_date'] = 'Nov 24 2014'\n",
      "run_info['experiment_name'] = 'Various Sample Info'\n",
      "run_info['output_dir'] = tscc_data+'/nov_24_run/'\n",
      "samples_dir = tscc_data+'/nov_24_run/fastq'\n",
      "sample_info = {}"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 56
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "with open(out_template, 'w') as OUT:\n",
      "    with open(run_template_yaml) as RUN:\n",
      "        for line in RUN:\n",
      "            for key, value in run_info.items():\n",
      "                line = line.replace(key, value)\n",
      "            OUT.write(line)\n",
      "    for i in range(0, len(sample_files), 2):\n",
      "        sample_info['file1'] = os.path.join(samples_dir, sample_files[i])\n",
      "        sample_info['file2'] = os.path.join(samples_dir, sample_files[i+1])\n",
      "        sample_info['sample_name'] = sample_files[i].split('_')[1]\n",
      "        with open(sample_template_yaml) as SAMPLE:\n",
      "            for line in SAMPLE:\n",
      "                for key, value in sample_info.items():\n",
      "                    line = line.replace(key, value)\n",
      "                OUT.write(line)\n",
      "        "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 57
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "!cat $out_template"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "fc_date: Nov 24 2014\r\n",
        "fc_name: Various Sample Info\r\n",
        "upload:\r\n",
        "  dir: /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/\r\n",
        "details:\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S10_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S10_L001_R2_001.fastq] \r\n",
        "    description: S10\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S11_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S11_L001_R2_001.fastq] \r\n",
        "    description: S11\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S12_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S12_L001_R2_001.fastq] \r\n",
        "    description: S12\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S13_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S13_L001_R2_001.fastq] \r\n",
        "    description: S13\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S14_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S14_L001_R2_001.fastq] \r\n",
        "    description: S14\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S15_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S15_L001_R2_001.fastq] \r\n",
        "    description: S15\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S16_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S16_L001_R2_001.fastq] \r\n",
        "    description: S16\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S17_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S17_L001_R2_001.fastq] \r\n",
        "    description: S17\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S18_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S18_L001_R2_001.fastq] \r\n",
        "    description: S18\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S19_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S19_L001_R2_001.fastq] \r\n",
        "    description: S19\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S1_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S1_L001_R2_001.fastq] \r\n",
        "    description: S1\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S20_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S20_L001_R2_001.fastq] \r\n",
        "    description: S20\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S21_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S21_L001_R2_001.fastq] \r\n",
        "    description: S21\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S22_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S22_L001_R2_001.fastq] \r\n",
        "    description: S22\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S23_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S23_L001_R2_001.fastq] \r\n",
        "    description: S23\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S2_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S2_L001_R2_001.fastq] \r\n",
        "    description: S2\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S3_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S3_L001_R2_001.fastq] \r\n",
        "    description: S3\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S4_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S4_L001_R2_001.fastq] \r\n",
        "    description: S4\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S5_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S5_L001_R2_001.fastq] \r\n",
        "    description: S5\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S6_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S6_L001_R2_001.fastq] \r\n",
        "    description: S6\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S7_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S7_L001_R2_001.fastq] \r\n",
        "    description: S7\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S8_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S8_L001_R2_001.fastq] \r\n",
        "    description: S8\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S9_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/HCN_S9_L001_R2_001.fastq] \r\n",
        "    description: S9\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n",
        "  - files: [/projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/Undetermined_S0_L001_R1_001.fastq, /projects/ps-jcvi/projects/Lasken_Single_Cell/data/nov_24_run/fastq/Undetermined_S0_L001_R2_001.fastq] \r\n",
        "    description: S0\r\n",
        "    genome_build: GRCh37ERCC\r\n",
        "    analysis: RNA-seq\r\n",
        "    algorithm:\r\n",
        "         aligner: tophat2\r\n",
        "         quality_format: Standard\r\n",
        "         trim_reads: read_through\r\n",
        "         adapters: [smartseq2, nextera]\r\n",
        "         strandedness: unstranded\r\n"
       ]
      }
     ],
     "prompt_number": 58
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "!ls -lh $local_data/nov_24_run/fastq"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "total 11G\r\n",
        "-rw-r--r-- 1 131998 7592 168M Nov 26 02:54 HCN_S10_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  13M Nov 30 20:50 HCN_S10_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 168M Nov 26 02:54 HCN_S10_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  49M Nov 30 20:51 HCN_S10_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 105M Nov 26 02:54 HCN_S11_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  28M Nov 30 20:51 HCN_S11_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 105M Nov 26 02:55 HCN_S11_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  31M Nov 30 20:51 HCN_S11_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 160M Nov 26 02:55 HCN_S12_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  44M Nov 30 20:51 HCN_S12_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 161M Nov 26 02:56 HCN_S12_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  51M Nov 30 20:51 HCN_S12_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 173M Nov 26 02:56 HCN_S13_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  46M Nov 30 20:51 HCN_S13_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 173M Nov 26 02:57 HCN_S13_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  52M Nov 30 20:51 HCN_S13_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 132M Nov 26 02:57 HCN_S14_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  35M Nov 30 20:51 HCN_S14_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 133M Nov 26 02:57 HCN_S14_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  39M Nov 30 20:51 HCN_S14_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 161M Nov 26 02:57 HCN_S15_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  39M Nov 30 20:51 HCN_S15_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 161M Nov 26 02:58 HCN_S15_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  46M Nov 30 20:51 HCN_S15_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 168M Nov 26 02:58 HCN_S16_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  40M Nov 30 20:51 HCN_S16_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 169M Nov 26 02:58 HCN_S16_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  47M Nov 30 20:51 HCN_S16_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 186M Nov 26 02:59 HCN_S17_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  46M Nov 30 20:51 HCN_S17_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 187M Nov 26 02:59 HCN_S17_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  54M Nov 30 20:51 HCN_S17_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 161M Nov 26 02:59 HCN_S18_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  44M Nov 30 20:51 HCN_S18_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 162M Nov 26 03:00 HCN_S18_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  50M Nov 30 20:51 HCN_S18_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 134M Nov 26 03:00 HCN_S19_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  36M Nov 30 20:51 HCN_S19_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 135M Nov 26 03:01 HCN_S19_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  44M Nov 30 20:51 HCN_S19_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592  50M Nov 26 02:46 HCN_S1_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  17M Nov 30 20:51 HCN_S1_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592  50M Nov 26 02:46 HCN_S1_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  15M Nov 30 20:51 HCN_S1_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592  23M Nov 26 03:01 HCN_S20_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592 7.7M Nov 30 20:51 HCN_S20_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592  23M Nov 26 03:01 HCN_S20_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592 6.2M Nov 30 20:51 HCN_S20_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 171M Nov 26 03:01 HCN_S21_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  40M Nov 30 20:51 HCN_S21_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 172M Nov 26 03:01 HCN_S21_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  46M Nov 30 20:51 HCN_S21_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 226M Nov 26 03:02 HCN_S22_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  61M Nov 30 20:51 HCN_S22_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 227M Nov 26 03:02 HCN_S22_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  63M Nov 30 20:52 HCN_S22_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 163M Nov 26 03:02 HCN_S23_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  45M Nov 30 20:51 HCN_S23_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 163M Nov 26 03:03 HCN_S23_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  49M Nov 30 20:52 HCN_S23_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 204M Nov 26 02:46 HCN_S2_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  52M Nov 30 20:51 HCN_S2_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 205M Nov 26 02:47 HCN_S2_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  60M Nov 30 20:51 HCN_S2_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 214M Nov 26 02:47 HCN_S3_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  58M Nov 30 20:51 HCN_S3_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 214M Nov 26 02:48 HCN_S3_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  63M Nov 30 20:52 HCN_S3_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 168M Nov 26 02:49 HCN_S4_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  46M Nov 30 20:51 HCN_S4_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 169M Nov 26 02:49 HCN_S4_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  53M Nov 30 20:52 HCN_S4_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 239M Nov 26 02:49 HCN_S5_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  62M Nov 30 20:51 HCN_S5_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 240M Nov 26 02:50 HCN_S5_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  71M Nov 30 20:52 HCN_S5_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 238M Nov 26 02:51 HCN_S6_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  59M Nov 30 20:51 HCN_S6_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 239M Nov 26 02:52 HCN_S6_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  69M Nov 30 20:52 HCN_S6_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 203M Nov 26 02:52 HCN_S7_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  53M Nov 30 20:51 HCN_S7_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 204M Nov 26 02:53 HCN_S7_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  60M Nov 30 20:52 HCN_S7_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 136M Nov 26 02:53 HCN_S8_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  39M Nov 30 20:51 HCN_S8_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 136M Nov 26 02:53 HCN_S8_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  41M Nov 30 20:52 HCN_S8_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 121M Nov 26 02:54 HCN_S9_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  32M Nov 30 20:51 HCN_S9_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 121M Nov 26 02:54 HCN_S9_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592  34M Nov 30 20:51 HCN_S9_L001_R2_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 613M Nov 26 03:03 Undetermined_S0_L001_R1_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592 165M Nov 30 20:52 Undetermined_S0_L001_R1_001.fastq.gz\r\n",
        "-rw-r--r-- 1 131998 7592 614M Nov 26 03:05 Undetermined_S0_L001_R2_001.fastq\r\n",
        "-rw-r--r-- 1 131998 7592 189M Nov 30 20:54 Undetermined_S0_L001_R2_001.fastq.gz\r\n"
       ]
      }
     ],
     "prompt_number": 60
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}